src/multiset.rs

```diff
@@ -315,8 +315,10 @@ impl<K> HashMultiSet<K> where
     /// assert!(multiset.count_of(&5) == 3);
     /// multiset.remove_all(&5);
     /// assert!(multiset.count_of(&5) == 0);
+    /// assert!(multiset.len() == 0);
     /// ```
     pub fn remove_all(&mut self, val: &K) {
+        self.size -= self.elem_counts.get(val).unwrap_or(&0);
```

Comment:
Nice catch! Thanks for adding the tests too.

Reply:
Oh, I see now this was from #21, but it made its way into the source branch for this PR too. Now I see why that other PR got automatically merged.

Thanks again :+1: